### ISSUE TITLE
Upgrade the AKS Ubuntu 18.04 image to use 18.0.4.5+

### ISSUE THREAD

**Issue by gldraphael** [Azure/AKS#2798]
**tl;dr:**

The nodes currently use Ubuntu 18.04.1. Upgrading the Host OS to 18.04.5 (or later) would fix a difficult to diagnose bug when working with Azure File Share mounts.

---

When a pod that mounts an Azure File Share stays inactive for a long time (about 3-4 days), the pod can no longer access the mount. It instead throws a Host is down error message.

```
OSError: [Errno 112] Host is down: '/path/to/file/in/mount'
```

According to the [Azure Files documentation](https://docs.microsoft.com/en-us/azure/storage/files/storage-troubleshoot-linux-file-connection-problems#mount-error112-host-is-down-because-of-a-reconnection-time-out):

> A "112" mount error occurs on the Linux client when the client has been idle for a long time. After an extended idle time, the client disconnects and the connection times out. The connection can be idle... [when] recent reconnection fixes are not present in older kernels

According to another page on the [Azure Files documentation](https://docs.microsoft.com/en-us/azure/storage/files/storage-how-to-use-files-linux?tabs=smb311), Azure Files requires Ubuntu 18.04.5 LTS+ for SMB 3.1.1 connections (and Ubuntu 16.04.4 LTS+ for 3.0).

---

**What happened**:
**What you expected to happen**:
**How to reproduce it (as minimally and precisely as possible)**:
**Anything else we need to know?**:

---

**Environment**:

- **Kubernetes version:** v1.20.7
- **Size of cluster:** 16 when idle
- **General description of workloads in the cluster:** the pod in question runs a python ML script

**Comment by miwithro** [comment]
@Alexeldeib

**Comment by gldraphael** [comment]
Update:

We upgraded our test cluster to kubernetes version 1.22.6 and created a nodepool that uses 
AKSUbuntu-1804gen2containerd-2022.02.07.

According to [this](https://github.com/Azure/AKS/blob/2022-02-10/vhd-notes/aks-ubuntu/AKSUbuntu-1804/2022.02.07.txt#L914), that image uses 18.04.1, but a `cat /etc/os-release` on the node returns 18.04.5 LTS.

Worth noting that the cluster where we are seeing the issue uses AKSUbuntu-1804gen2containerd-2021.08.14 & `cat /etc/os-release` on these nodes also return 18.04.5 LTS (Bionic Beaver).

I will try to reproduce this on the new nodepool and report back by the end of next week.

**Comment by gldraphael** [comment]
Am no SMB expert, but it _could_ have to do with the kernel version like @datwelk suggested.

However we do not use a premium Azure File share. We're not really trying to use SMB Multichannel.

**Comment by alexeldeib** [comment]
The link you shared dumps uname -a at the end of build, but at the end of the day it's whatever base the Ubuntu marketplace image publishes. And as you realized, components in that build are much newer than 18.04.1 -- it does have the 5.4 kernel for example. 

I suggested to @miwithro to check with linux/canonical folks internally to verify. It would be good to know which specific requirement forces 18.04.5 and whether or not the Azure images already have that in the base.

**Comment by gldraphael** [comment]
I cannot reproduce this on the updated cluster.

<details>
<summary>Details</summary>

Created a pod as follows:

```yaml
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: azurefile
provisioner: kubernetes.io/azure-file
mountOptions:
  - dir_mode=0777
  - file_mode=0777
  - uid=1000
  - gid=1000
  - mfsymlinks
  - cache=strict
parameters:
  skuName: Standard_LRS
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: smbexp-pvc
  namespace: experiments
spec:
  accessModes:
    - ReadWriteMany
  storageClassName: azurefile
  resources:
    requests:
      storage: 128Gi
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: smbexp
  namespace: experiments
  labels:
    app: smbexp
spec:
  replicas: 1
  strategy:
    type: Recreate
  selector:
    matchLabels:
      app: smbexp
  template:
    metadata:
      labels:
        app: smbexp
    spec:
      containers:
        - name: galdin
          image: ubuntu:latest
          command: [ "sleep" ]
          args: [ "infinity" ]
          resources:
            requests:
              cpu: 200m
              memory: 512Mi
            limits:
              cpu: "2"
              memory: 2Gi
          volumeMounts:
            - mountPath: "/smb"
              name: volume
      volumes:
        - name: volume
          persistentVolumeClaim:
            claimName: smbexp-pvc
```

The pod has been running for close to 6 days.
I execed into the pod, and could confirm that the smb drive works.

```sh
kubectl exec -it smbexp-59cf6dbd5c-45b8s -n experiments -- /bin/bash
root@smbexp-59cf6dbd5c-45b8s:/# ls /smb/
hello
root@smbexp-59cf6dbd5c-45b8s:/# ls /smb/hello/
hello
root@smbexp-59cf6dbd5c-45b8s:/# cat /smb/hello/hello
world
root@smbexp-59cf6dbd5c-45b8s:/#
```

On the old cluster I see a `Host is down` error even when attempting to `ls` the contents of `/smb`.

</details>
